packages/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,256 @@
+import React, {
+  useEffect,
+  useState,
+  useRef,
+  useMemo,
+  Fragment,
+  ReactNode,
+  RefObject,
+  ReactElement,
+} from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import debounce from 'lodash/lodash';
+import { emotion } from '@leafygreen-ui/lib';
+import { calculatePosition, getElementPosition } from './positionUtils';
+
+const { css, cx } = emotion;
+
+const rootPopoverStyle = css`
+  transition: transform 150ms ease-in-out, opacity 150ms ease-in-out;
+  position: absolute;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+/**
+ * Options to determine the alignment of the popover relative to
+ * the other component
+ * ---
+ * @param Top will align content above other element
+ * @param Bottom will align content below other element
+ * @param Left will align content to the left of other element
+ * @param Right will align content to the right of other element
+ */
+export enum Align {
+  Top = 'top',
+  Bottom = 'bottom',
+  Left = 'left',
+  Right = 'right',
+}
+
+/**
+ * Options to determine the justification of the popover relative to
+ * the other component
+ * ---
+ * @param Start will justify content against the start of other element
+ * @param Middle will justify content against the middle of other element
+ * @param Bottom will justify content against the end of other element
+ */
+export enum Justify {
+  Start = 'start',
+  Middle = 'middle',
+  End = 'end',
+}
+
+function getWindowSize() {
+  return {
+    width: window.innerWidth,
+    height: window.innerHeight,
+  };
+}
+
+function useWindowResize() {
+  const [windowSize, setWindowUpdateVal] = useState(getWindowSize());
+
+  function calcResize() {
+    debounce(setWindowUpdateVal(getWindowSize()));
+  }
+
+  useEffect(() => {
+    window.addEventListener('resize', calcResize);
+
+    return () => window.removeEventListener('resize', calcResize);
+  });
```

Comment:
We want this to only run on initial mount, not on every re-render. So we need to add a second argument of an empty array to skip additional renders. Docs: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects